server/src/main/java/org/elasticsearch/search/aggregations/bucket/filter/FiltersAggregator.java

```diff
@@ -61,7 +61,7 @@ public abstract class FiltersAggregator extends BucketsAggregator {
 
     public static class KeyedFilter implements Writeable, ToXContentFragment {
         private final String key;
-        private final QueryBuilder filter;
```

Comment:
Unfortunately, I had to make the filter here mutable, because the query for API Key aggs has to be modified to go only over API Key docs, and also optionally of a single user.